src/openzeppelin/account/library.cairo

```diff
@@ -163,32 +199,47 @@ namespace Account:
 
         let (__fp__, _) = get_fp_and_pc()
         let (tx_info) = get_tx_info()
-        let (_current_nonce) = Account_current_nonce.read()
+        let (local ecdsa_ptr : SignatureBuiltin*) = alloc()
+        with ecdsa_ptr:
+            # validate transaction
+            let (is_valid) = is_valid_signature(tx_info.transaction_hash, tx_info.signature_len, tx_info.signature)
```

Comment:
I believe this line should be part of the error message `("Account: invalid signature")` because the method `is_valid_signature` will revert without message when the signature is not valid .